ambry-router/src/main/java/com.github.ambry.router/DeleteOperation.java

```diff
@@ -99,10 +101,17 @@
    *                            that gets created as part of this poll operation.
    */
   void poll(RequestRegistrationCallback<DeleteOperation> requestRegistrationCallback) {
-    cleanupExpiredInflightRequests();
-    checkAndMaybeComplete();
-    if (!isOperationComplete()) {
-      fetchRequests(requestRegistrationCallback);
+    try {
+      cleanupExpiredInflightRequests();
+      checkAndMaybeComplete();
+      if (!isOperationComplete()) {
+        fetchRequests(requestRegistrationCallback);
+      }
+    } catch (Exception e) {
+      logger.error("Unexpected error in DeleteBlobOperation poll; operation cancelled.", e);
```

Comment:
a general rule of thumb to follow i think would be to not log unless the exception is being swallowed or it is the very last destination of the exception. In this case, the exception is being set and will be passed to someone else who will log it eventually.

If you log when the exception is not being swallowed, there might be double logging and pollution of the log.